pkg/datastore/database.go

```diff
@@ -240,25 +254,37 @@ func (db *relationalDb) FindInTable(ctx context.Context, tableName string, recor
 		}
 	}()
 
-	if err = tx.Table(tableName).Where(record).First(record).Error; err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return ErrRecordNotFound.Wrap(err).WithValue("record", fmt.Sprintf("%+v", record)).WithValue(DB_NAME, db.dbName)
+	if softDelete {
```

Comment:
you could just use softDelete for setting tx := tx.Unscoped()

Reply:
If tx is carried inside ctx, tx := tx.Unscoped() will make all following functions calls to have tx.Unscoped(), which creates errors

Reply:
tx can be a new variable too you are don't need to override

Reply:
@YanhaoLi07 resolving for now to unblock Gaurav, please follow up with a change